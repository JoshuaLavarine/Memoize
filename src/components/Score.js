import React, {Component} from 'react';

class Score extends Component {
  constructor() {
    super();
    this.state = {
      // empty
    }
  }

  // selectPioneerAndResetIncorrectAnswers = () => {
  //   this.props.selectPioneer()
  //   this.props.resetIncorrectAnswers()
  // }

  render() {
    let currentPioneer = this.props.pioneers.pioneersData.find(pioneer => {
      return pioneer.id == this.props.currentPioneer
    })
    switch(this.props.currentQuestionIndex === this.props.arrayLength.length) {
      case(true):
      return (
        <section className="article-layout-correct">
            <p>You scored {this.props.guessesCorrect} out of {this.props.currentUserGuess}</p>
            <section className="button-container-correct">
              <button className="button-quiz-correct" onClick={this.props.selectPioneer} id={0}>Choose New Pioneer</button>
            </section>
        </section>
      )
    default:
      return (
        <section className="article-layout">
          <p>You scored {this.props.guessesCorrect} out of {this.props.currentUserGuess}</p>
           <section className="button-container">
            <button className="button-back" onClick={this.props.selectPioneer} value={0}>Return to Home Screen</button>
            <button className="button-quiz" onClick={this.props.displayScore}>Next Question</button>
          </section> 
        </section>
      )
    }
  }
}

export default Score;