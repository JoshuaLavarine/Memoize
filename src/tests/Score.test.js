import React from "react";
import Score from "../Components/Score";
import { shallow } from "enzyme";
import { pioneers } from "./mockData";

describe("Score", () => {
  const displayScore = jest.fn();
  const selectPioneer = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Score 
      guessesCorrect = {2}
      displayScore = {displayScore}
      currentPioneer = {1}
      selectPioneer = {selectPioneer}
      pioneers = {pioneers}
      currentUserGuess = {1}
      currentQuestionIndex = {0}
      incorrectAnswers = {[1, 2, 3, 4, 5]}
      />   
    );
  });
	
  it("should match the snapshot with all data passed in", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should match the alternative snapshot with all data passed in", () => {
    const newWrapper = shallow(    
      <Score 
        guessesCorrect = {2}
        displayScore = {displayScore}
        currentPioneer = {1}
        selectPioneer = {selectPioneer}
        pioneers = {pioneers}
        currentUserGuess = {1}
        currentQuestionIndex = {1}
        incorrectAnswers = {[1]}
      /> 
    )
    expect(newWrapper).toMatchSnapshot();
  });
});