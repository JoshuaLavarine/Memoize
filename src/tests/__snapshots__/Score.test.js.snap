// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score should match the alternative snapshot with all data passed in 1`] = `
<section
  className="article-layout-correct"
>
  <p>
    You scored 
    2
     out of 
    1
  </p>
  <section
    className="button-container-correct"
  >
    <button
      className="button-quiz-correct"
      id={0}
      onClick={[MockFunction]}
    >
      Choose New Pioneer
    </button>
  </section>
</section>
`;

exports[`Score should match the snapshot with all data passed in 1`] = `
<section
  className="article-layout"
>
  <p>
    You scored 
    2
     out of 
    1
  </p>
  <section
    className="button-container"
  >
    <button
      className="button-back"
      onClick={[MockFunction]}
      value={0}
    >
      Return to Home Screen
    </button>
    <button
      className="button-quiz"
      onClick={[MockFunction]}
    >
      Next Question
    </button>
  </section>
</section>
`;
